tests/transactions/tests.py

```diff
@@ -483,12 +483,15 @@ def test_orm_query_after_error_and_rollback(self):
         mode (#27504).
         """
         transaction.set_autocommit(False)
-        r1 = Reporter.objects.create(first_name='Archibald', last_name='Haddock')
-        r2 = Reporter(first_name='Cuthbert', last_name='Calculus', id=r1.id)
-        with self.assertRaises(IntegrityError):
-            r2.save(force_insert=True)
-        transaction.rollback()
-        Reporter.objects.last()
+        try:
+            r1 = Reporter.objects.create(first_name='Archibald', last_name='Haddock')
+            r2 = Reporter(first_name='Cuthbert', last_name='Calculus', id=r1.id)
+            with self.assertRaises(IntegrityError):
+                r2.save(force_insert=True)
+            transaction.rollback()
+            Reporter.objects.last()
+        finally:
+            transaction.set_autocommit(True)
```

Comment:
We need to turn `autocommit` back on.

Reply:
Would it be cleaner to do this with `setUp()` and `tearDown()` like this class elsewhere in the module? https://github.com/django/django/blob/7bb0605a8c4adb4247a179276036f138ae859ac9/tests/transactions/tests.py#L228-L235
(The other test in the class seems to need it, too.)

Reply:
True, thanks for the suggestion :+1: